plugins/performance-lab/includes/admin/plugin-activate-ajax.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Handles activation of Performance Features (Plugins) using AJAX.
+ */
+
+/* global perflabPluginActivateAjaxData */
+( function () {
+	// @ts-ignore
+	const { i18n, a11y } = wp;
+	const { __ } = i18n;
+
+	/**
+	 * Handles click events on elements with the class 'perflab-install-active-plugin'.
+	 *
+	 * This asynchronous function listens for click events on the document and executes
+	 * the provided callback function if triggered.
+	 *
+	 * @param {MouseEvent} event - The click event object that is triggered when the user clicks on the document.
+	 *
+	 * @return {Promise<void>} - The asynchronous function returns a promise that resolves to void.
+	 */
+	async function handlePluginActivationClick( event ) {
+		const target = /** @type {HTMLElement} */ ( event.target );
+
+		if ( ! target.classList.contains( 'perflab-install-active-plugin' ) ) {
+			return;
+		}
+
+		// Prevent the default link behavior.
+		event.preventDefault();
+
+		target.classList.add( 'updating-message' );
+		target.textContent = __( 'Activating…', 'performance-lab' );
+		target.style.pointerEvents = 'none';
+		target.parentElement.style.cursor = 'not-allowed';
+
+		a11y.speak( __( 'Activating…', 'performance-lab' ) );
+
+		const pluginSlug = target.getAttribute( 'data-plugin-slug' ).trim();
+
+		// Send an AJAX POST request to activate the plugin.
+		try {
+			const response = await fetch(
+				// @ts-ignore
+				perflabPluginActivateAjaxData.ajaxUrl,
+				{
+					method: 'POST',
+					credentials: 'same-origin',
+					headers: {
+						'Content-Type': 'application/x-www-form-urlencoded',
+					},
+					body: new URLSearchParams( {
+						action: 'perflab_install_activate_plugin',
+						slug: pluginSlug,
+						// @ts-ignore
+						_ajax_nonce: perflabPluginActivateAjaxData.nonce,
+					} ),
+				}
+			);
+
+			const responseData = await response.json();
+
+			if ( ! responseData.success ) {
+				target.classList.remove( 'updating-message' );
+				target.textContent = __( 'Activate', 'performance-lab' );
+				target.style.pointerEvents = '';
+				target.parentElement.style.cursor = '';
+
+				return;
+			}
+
+			const newButton = document.createElement( 'button' );
+
+			newButton.type = 'button';
+			newButton.className = 'button button-disabled';
+			newButton.disabled = true;
+			newButton.textContent = __( 'Active', 'performance-lab' );
+			target.parentElement.style.cursor = '';
+
+			target.parentNode.replaceChild( newButton, target );
```

Comment:
Maybe just update the existing button instead of replace the button? By keeping the same element in place, users with a keyboard won't have the focus removed from the document.

Reply:
The target is actually `<a>` tag ( because it should be able to work even if JavaScript is disabled ) and to match the mark up of loaded page which is disabled button because of that I am replacing the target `<a>` tag with button.

Reply:
I think the `disabled` class works with a link as well. I think you can keep it as a link instead of replacing it with a button.